.github/workflows/CI.yml

```diff
@@ -0,0 +1,64 @@
+name: Tests
+
+on:
+  pull_request:
```

Comment:
you have to add branches here for pull_request

Reply:
i don't think that's correct. the `branches` tag is a filter- when present the trigger will only occur on those branches. when no filter is specified it will trigger on any pull request. I'll see if i can find a reference

Reply:
I think that's got more to do with the perms in this repo. You can see that the workflow is triggering in my fork.
Do you want to take a look at the repo settings?

Reply:
https://github.com/joeirimpan/zammad_py/pull/154/files
^ i added the [master, develop] into pull_request section and it triggered here. Can you add that here and we are good to go?

Reply:
GH triggered on your fork because it was push.

Reply:
Sorry, i still don't think that's right.

If you look at the job on my fork here - https://github.com/danieleades/zammad_py/actions/runs/1858052301
you can see that the job was triggered by `pull_request`, not `push` (check the text in the top left of the job window).

by pushing a version yourself you've just done the same test that i've done in my fork. I don't think it demonstrates that the job on this PR isn't triggering because of the missing `branches` field.

I have to admit I don't fully understand why the workflow isn't triggering on this PR. Could be that you need a commit in the default branch with a workflow before they are triggered for the repo, or it could be that you need to update the settings to allow PRs from forks to trigger workflows (they are *not* enabled by default). I can't see the actions permissions for your repo, you might want to check that these are enabled in settings (`settings` > `actions` > `general` > `actions permissions`). It's possible these are configured to only allow "local" actions to run, as opposed to actions which live in a fork.

i can show you some examples of repos using this pattern where the workflows *are* triggered on PRs. It's not an uncommon pattern.

- https://github.com/psf/requests/blob/main/.github/workflows/run-tests.yml
- https://github.com/useblocks/sphinxcontrib-needs/blob/master/.github/workflows/ci.yaml
- https://github.com/danieleades/monzo-lib/blob/main/.github/workflows/CI.yml
- https://github.com/python-poetry/poetry-core/blob/master/.github/workflows/code-quality.yaml
- https://github.com/python-poetry/poetry/blob/master/.github/workflows/main.yml
- etc.

Reply:
ah okay. I can see action permission is set to `Allow all actions`. I haven't played around with gh action much so i might be wrong. Let me try merging this patch and see if its triggering.